### ISSUE TITLE
 Unsupported PNG color type: 3

### ISSUE THREAD

**Issue by gustavla** [apple/turicreate#345]
Some PNG images can't be loaded (I'm not sure :

```python
>>> img = tc.Image('bike_261.mask.0.png')
turicreate.toolkits._main.ToolkitError: Unsupported PNG color type: 3
```

I'm not sure what kind of image this is (I think a palette with three fixed images), but here's some more info:

```
bike_261.mask.0.png PNG 640x480 640x480+0+0 8-bit sRGB 3c 2258B 0.000u 0:00.000
```

An image that triggers this can be found [here](https://lear.inrialpes.fr/people/marszalek/data/ig02/) (the image in the example is from the bikes zip file)

**Comment by ylow** [comment]
In png_io.cpp, we process only a subset of the png format
```
  channels = png_num_channels(color_type);
  if (channels != 1 && channels != 3 && channels != 4) {
    png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
    log_and_throw(std::string("Unsupported PNG color type: ") + std::to_string(color_type));
  }
```
Specifically, if color_type = 3 (indexed palette), channels is -1.

I think we can make the png reader a bit more general and possibly more performant as well. (Use png_read_image rather than png_read_row)

**Comment by gustavla** [comment]
I can't seem to reproduce this on 4.2, so I'm closing this. I think it was fixed in #278.
